main.cpp

```diff
@@ -218,10 +224,15 @@
             rtld_fini,
             stack_end
         );
-        if(stateThreadRunning){
-            _DEBUG("Waiting for thread to exit");
-            exitThread = true;
-            stateThread.join();
+        exitThread = true;
```

Comment:
As only one place sets the value, and all other locations read the value, what kind of data race could potentially happen here that would result in unexpected behaviour?

Reply:
Since the places reading this value are in an infinite loop, where if it missed the changed value, it would just catch it on the next loop, and the main thread just waits for that thread to finish after setting this value, is the data race something to actually be concerned about here?